python/across_api/base/fov.py

```diff
@@ -0,0 +1,382 @@
+
+        # Create a HEALPix map of the probability density distribution
+        prob = healpix_map_from_position_error(
+            skycoord=skycoord, error_radius=error_radius, nside=nside
+        )
+
+        return self.in_fov_healpix_map(healpix_loc=prob)
+
+    def in_fov_healpix_map(
+        self,
+        healpix_loc: FITS_rec,
+        healpix_order: str = "NESTED",
+    ) -> float:
+        """
+        Calculates the amount of probability inside the field of view (FOV)
+        defined by the given parameters. This works by calculating a SkyCoord
+        containing every non-zero probability pixel, uses the
+        `in_fov` method to check which pixels are inside the FOV,
+        and then finding the integrated probability of those pixels.
+
+        Note: This method makes no attempt to deal with pixels that are only
+        partially inside the FOV, i.e. Earth occultation is calculated for
+        location of the center of each HEALPix pixel.
+
+        If `healpix_order` == "NUNIQ", it assumes that `healpix_loc` contains a
+        multi-order HEALPix map, and handles that accordingly.
+
+        Parameters
+        ----------
+        healpix_loc
+                An array containing the probability density values for each HEALPix
+                pixel.
+        healpix_nside
+                The NSIDE value of the HEALPix map. If not provided, it will be
+                calculated based on the length of healpix_loc.
+        healpix_order
+                The ordering scheme of the HEALPix map. Default is "NESTED".
+
+        Returns
+        -------
+        float
+                The amount of probability inside the FOV.
+
+        """
+        # Extract the NSIDE value from the HEALPix map, also level and ipix if
+        # this is a MOC map
+        if healpix_order == "NUNIQ":
+            level, ipix = ah.uniq_to_level_ipix(healpix_loc["UNIQ"])
+            uniq_nside = ah.level_to_nside(level)
+            healpix_loc = healpix_loc["PROBDENSITY"]
+        else:
+            nside = ah.npix_to_nside(len(healpix_loc))
+
+        # Find where in HEALPix map the probability is > 0
+        nonzero_prob_pixels = np.where(healpix_loc > 0.0)[0]
+
+        # Create a list of RA/Dec coordinates for these pixels
+        if healpix_order == "NUNIQ":
+            ra, dec = ah.healpix_to_lonlat(
+                ipix[nonzero_prob_pixels],
+                nside=uniq_nside[nonzero_prob_pixels],  # type: ignore
+                order="NESTED",
+            )
+        else:
+            ra, dec = ah.healpix_to_lonlat(
+                nonzero_prob_pixels, nside=nside, order=healpix_order
+            )
+
+        # Convert these coordinates into a SkyCoord
+        skycoord = SkyCoord(ra=ra, dec=dec, unit="deg")
+
+        # Calculate pixel indicies of the all the regions inside of the FOV
+        visible_probability_pixels = nonzero_prob_pixels[self.in_fov(skycoord=skycoord)]
```

Comment:
This will be inefficient if the HEALPix map is flat-resolution and matches the resolution used for the FOV intersection test, and it will be incorrect if the HEALPix map is multi-resolution or coarser in resolution than the FOV intersection test.

Reply:
This is still incorrect, or at least inaccurate. What you should do is upsample both HEALPix maps to the higher of the two resolutions and then sum. That would be faster and more accurate.

There should be no need to do any sky coordinate to HEALPix conversion here. Upsampling a HEALPix map just means repeating every pixel (nside_new / nside_old)**2 times, whereas these coordinate conversions involve unnecessary trigonometry.

Reply:
We discussed that this works _well enough_ at the moment, we can create an issue on this after the BurstCube MVP